docs/non-functional-requirements/portability.md

```diff
@@ -0,0 +1,35 @@
+# Portability
+
+Portability refers to the ease with which software can be transferred and used in different environments or platforms without requiring significant modification. This includes moving the software across various hardware, operating systems, cloud services, or development frameworks while maintaining its functionality, performance, and usability.
+
+## Characteristics
+
+- Platform Independence: The ability of the software to run on different operating systems, hardware architectures, and devices without requiring major changes.
+- Minimal Modification: The need for minimal code changes or reconfiguration when moving the software to a different environment.
+- Standard Compliance: Adherence to industry standards and protocols to ensure compatibility across different systems and platforms.
+- Environment Abstraction: Use of abstraction layers or frameworks that isolate the software from specific platform details, making it easier to adapt to different environments.
+- Configuration Flexibility: Ease of modifying configuration settings to suit different environments without altering the core software code.
+- Dependency Management: Efficient handling of external dependencies, ensuring that required libraries, tools, and services are available or can be easily obtained in the new environment.
+- Packaging and Distribution: Efficient packaging methods, such as containerization (e.g., Docker), that encapsulate the software and its dependencies to facilitate deployment in diverse environments.
+- Modular Design: Designing the software in a modular way, where components can be independently developed, tested, and deployed, enhancing the ease of porting parts of the system.
+
+## Implementations
+
+### Containerization
+
+- Docker: Packaging applications and their dependencies into containers, ensuring consistent behavior across different environments.
+- Kubernetes: Orchestrating containerized applications for deployment across various cloud providers and on-premises infrastructures.
+
+### Virtual Machines
+
+- Java Virtual Machine (JVM): Writing software in Java or other JVM languages to run on any system with a compatible JVM.
+- VirtualBox or VMware: Using virtual machines to create consistent runtime environments regardless of the underlying hardware.
+
+### Platform-Agnostic Languages
+
+- Python, JavaScript, and Go: Utilizing programming languages known for their cross-platform capabilities to ensure code runs on multiple operating systems with little to no modification.
```

Comment:
About this, I would add a section about selecting the right programming language for you... 
Putting it out there that the language needs to serve you and no the other way around. 
The fact every service in a company is written in java, and you are a DS who's using numpy, doesn't mean you need to re-implment numpy on java

Reply:
Sure! done